providers/flagd/pkg/configuration.go

```diff
@@ -0,0 +1,116 @@
+package flagd
+
+import (
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/open-feature/go-sdk-contrib/providers/flagd/pkg/cache"
+	"os"
+	"strconv"
+)
+
+// Naming and defaults must comply with flagd environment variables
+const (
+	defaultMaxCacheSize                               int    = 1000
+	defaultPort                                              = 8013
+	defaultMaxEventStreamRetries                             = 5
+	defaultTLS                                        bool   = false
+	defaultCache                                      string = "lru"
+	defaultHost                                              = "localhost"
+	flagdHostEnvironmentVariableName                         = "FLAGD_HOST"
+	flagdPortEnvironmentVariableName                         = "FLAGD_PORT"
+	flagdTLSEnvironmentVariableName                          = "FLAGD_TLS"
+	flagdSocketPathEnvironmentVariableName                   = "FLAGD_SOCKET_PATH"
+	flagdServerCertPathEnvironmentVariableName               = "FLAGD_SERVER_CERT_PATH"
+	flagdCacheEnvironmentVariableName                        = "FLAGD_CACHE"
+	flagdMaxCacheSizeEnvironmentVariableName                 = "FLAGD_MAX_CACHE_SIZE"
+	flagdMaxEventStreamRetriesEnvironmentVariableName        = "FLAGD_MAX_EVENT_STREAM_RETRIES"
+	cacheDisabledValue                                       = cache.DisabledValue
+	cacheLRUValue                                            = cache.LRUValue
+	cacheInMemValue                                          = cache.InMemValue
+)
+
+type providerConfiguration struct {
+	CacheType                        string
+	CertificatePath                  string
+	EventStreamConnectionMaxAttempts int
+	Host                             string
+	MaxCacheSize                     int
+	OtelIntercept                    bool
+	Port                             uint16
+	SocketPath                       string
+	TLSEnabled                       bool
+}
+
+func newDefaultConfiguration() *providerConfiguration {
+	return &providerConfiguration{
+		CacheType:                        defaultCache,
+		EventStreamConnectionMaxAttempts: defaultMaxEventStreamRetries,
+		Host:                             defaultHost,
+		MaxCacheSize:                     defaultMaxCacheSize,
+		Port:                             defaultPort,
+		TLSEnabled:                       defaultTLS,
+	}
+}
+
+// updateFromEnvVar is a utility to update configurations based on current environment variables
+func (cfg *providerConfiguration) updateFromEnvVar(logger logr.Logger) {
+	portS := os.Getenv(flagdPortEnvironmentVariableName)
+	if portS != "" {
+		port, err := strconv.Atoi(portS)
+		if err != nil {
+			logger.Error(err,
+				fmt.Sprintf(
+					"invalid env config for %s provided, using default value: %d",
+					flagdPortEnvironmentVariableName, defaultPort,
+				))
+		} else {
+			cfg.Port = uint16(port)
+		}
+	}
+
+	host := os.Getenv(flagdHostEnvironmentVariableName)
+	if host != "" {
+		cfg.Host = host
+	}
```

Comment:
nit. This way we scope `host` to the bare minimum
```suggestion
	if host := os.Getenv(flagdHostEnvironmentVariableName); host != "" {
		cfg.Host = host
	}
```

Reply:
Updated all if conditions to use the same format